src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/Rfc2898DeriveBytes.OneShot.cs

```diff
@@ -0,0 +1,128 @@
+            byte[] password,
+            byte[] salt,
+            int iterations,
+            int length,
+            HashAlgorithmName hashAlgorithm)
+        {
+            if (password is null)
+                throw new ArgumentNullException(nameof(password));
+            if (salt is null)
+                throw new ArgumentNullException(nameof(salt));
+
+            return Pbkdf2DeriveBytes(new ReadOnlySpan<byte>(password), new ReadOnlySpan<byte>(salt), iterations, length, hashAlgorithm);
+        }
+
+        public static byte[] Pbkdf2DeriveBytes(
+            ReadOnlySpan<byte> password,
+            ReadOnlySpan<byte> salt,
+            int iterations,
+            int length,
+            HashAlgorithmName hashAlgorithm)
+        {
+            if (length < 0)
+                throw new ArgumentOutOfRangeException(nameof(length), SR.ArgumentOutOfRange_NeedNonNegNum);
+
+            byte[] result = new byte[length];
+            Pbkdf2DeriveBytes(password, salt, iterations, hashAlgorithm, result);
+            return result;
+        }
+
+        public static void Pbkdf2DeriveBytes(
+            ReadOnlySpan<byte> password,
+            ReadOnlySpan<byte> salt,
+            int iterations,
+            HashAlgorithmName hashAlgorithm,
+            Span<byte> destination)
+        {
+            Pbkdf2DeriveBytesCore(password, salt, iterations, hashAlgorithm, destination);
+        }
+
+        public static byte[] Pbkdf2DeriveBytes(
+            string password,
+            byte[] salt,
+            int iterations,
+            int length,
+            HashAlgorithmName hashAlgorithm)
+        {
+            if (password is null)
+                throw new ArgumentNullException(nameof(password));
+            if (salt is null)
+                throw new ArgumentNullException(nameof(salt));
+
+            return Pbkdf2DeriveBytes(password.AsSpan(), new ReadOnlySpan<byte>(salt), iterations, length, hashAlgorithm);
+        }
+
+        public static byte[] Pbkdf2DeriveBytes(
+            ReadOnlySpan<char> password,
+            ReadOnlySpan<byte> salt,
+            int iterations,
+            int length,
+            HashAlgorithmName hashAlgorithm)
+        {
+            if (length < 0)
+                throw new ArgumentOutOfRangeException(nameof(length), SR.ArgumentOutOfRange_NeedNonNegNum);
+
+            byte[] result = new byte[length];
+            Pbkdf2DeriveBytes(password, salt, iterations, hashAlgorithm, result);
+            return result;
+        }
+
+        public static void Pbkdf2DeriveBytes(
+            ReadOnlySpan<char> password,
+            ReadOnlySpan<byte> salt,
+            int iterations,
+            HashAlgorithmName hashAlgorithm,
+            Span<byte> destination)
+        {
+            byte[] passwordBytes = CryptoPool.Rent(Encoding.UTF8.GetMaxByteCount(password.Length));
+            int passwordBytesWritten = Encoding.UTF8.GetBytes(password, passwordBytes);
```

Comment:
@bartonjs is there "danger" here that we want to mitigate? e.g. something that results in invalid UTF8 so the bytes that come out are bogus?

Reply:
Not that I can think of.  The Encoding.UTF8 shared object is in the "map gibberish to the unknown character" mode.  Consistent garbage in produces consistent garbage out.